### PR TITLE
Query : Get pool status

### PR DESCRIPTION
**Pull request by parv0888**
depends on #94 

## Purpose
- Added support for query `GetPoolStatus`

## Changes
- Added support for query `GetPoolStatus`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolStatus.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import lombok.Data;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.Optional;
+
+@Data
+@Getter
+@ToString
+public class BakerPoolStatus extends PoolStatus {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolStatus.java:18]: Make fields final please. 
I don't see the need for `@Getter` `@ToString` when `@Data` is used.

File: README.md [hunk 2]
```diff
@@ -385,6 +391,18 @@ val bakerList = client
                 .getBakerList(Hash.from("2f15e174a42ec63d68abd8597e69573cf83199aacbfb9dae03c255d35b84aafb"));
 ```
 
+#### getPoolStatus
+```java
+val poolStatusPassiveDeletation = client
```
Comment (EmilLa1) [inline README.md:396]: specify types here.

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 3]
```diff
@@ -330,6 +333,23 @@ public ImmutableList<BakerId> getBakerList(Hash blockHash) throws BlockNotFoundE
...
+     * @param bakerId   {@link Optional<BakerId>} of the Baker of which pool is to be retrieved.
+     * @return Parsed {@link PoolStatus}.
+     */
+    public PoolStatus getPoolStatus(final Hash blockHash, final Optional<BakerId> bakerId) {
+        val res = server().getPoolStatus(ConcordiumP2PRpc.GetPoolStatusRequest.newBuilder()
+                .setBlockHash(blockHash.asHex())
+                .setPassiveDelegation(!bakerId.isPresent())
+                .setBakerId(bakerId.orElse(BakerId.from(0)).getId().getIndex().getValue())
+                .build());
+
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:349]: Can the result be `null` ? If so please act appropriately. 

I suggest something like `AccountNotFoundException`.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolOpenStatus.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.concordium.sdk.responses.poolstatus;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolOpenStatus.java:1]: There already exists a `OpenStatus.java`.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/CurrentPaydayStatus.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import com.concordium.sdk.transactions.CCDAmount;
+import com.concordium.sdk.types.UInt64;
+import lombok.Data;
+import lombok.Getter;
+import lombok.ToString;
+
+@Data
+@ToString
+@Getter
+public class CurrentPaydayStatus {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/CurrentPaydayStatus.java:12]: Make fields final please.
I don't see the need for @Getter @ToString when @Data is used.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PassiveDelegationStatus.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import concordium.ConcordiumP2PRpc;
+import lombok.Data;
+import lombok.Getter;
+import lombok.ToString;
+
+@Data
+@Getter
+@ToString
+public class PassiveDelegationStatus extends PoolStatus {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PassiveDelegationStatus.java:15]: Make fields final please.
I don't see the need for @Getter @ToString when @Data is used.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeReduceBakerCapital.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Data;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.Date;
+
+@Getter
+@ToString
+@Data
+public class PendingChangeReduceBakerCapital extends PendingChange {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeReduceBakerCapital.java:13]: Make fields final please.
I don't see the need for @Getter @ToString when @Data is used.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChange.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import lombok.Data;
+import lombok.Getter;
+import lombok.ToString;
+
+@Getter
+@Data
+@ToString
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChange.java:11]: Why `@Data` ?

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeNoChange.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package com.concordium.sdk.responses.poolstatus;
+
+import lombok.Data;
+import lombok.Getter;
+import lombok.ToString;
+
+@Data
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeNoChange.java:7]: Why `@Data`?

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeRemovePool.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+@Getter
+@ToString
+@Data
+public class PendingChangeRemovePool extends PendingChange {
+
+    /**
+     * Effective time of the change.
+     */
+    private Date effectiveTime;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeRemovePool.java:17]: Please make final. 
And I don't see the need for `@ToString` and `@Getter` when `@Data` is being used.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolInfo.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package com.concordium.sdk.responses.poolstatus;
+
+import com.concordium.sdk.responses.accountinfo.CommissionRates;
+
+/**
+ * Additional information about a baking pool. This information is added with the introduction of delegation.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolInfo.java:6]:
```suggestion
 * Additional information about a baking pool. This information is available from P4.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.Getter;
+import lombok.ToString;
+import lombok.val;
+
+@Data
+@Getter
+@ToString
+public class PoolStatus {
+
+    private PoolType poolType;
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java:16]: Please make final

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import concordium.ConcordiumP2PRpc;
+import lombok.Data;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.val;
+
+@Data
+@Getter
+@ToString
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java:13]: I don't see the need for `@Getter` and `@ToString` when `@Data` is being used.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolType.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
+package com.concordium.sdk.responses.poolstatus;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public enum PoolType {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolType.java:6]: Please add documentation.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]

**Comment by EmilLa1** [comment]
Also please add a unit test.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/CurrentPaydayStatus.java (hunk 5 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/CurrentPaydayStatus.java:12]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeNoChange.java (hunk 9 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeNoChange.java:7]: removed

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeReduceBakerCapital.java (hunk 7 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeReduceBakerCapital.java:13]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeRemovePool.java (hunk 10 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PendingChangeRemovePool.java:17]: done

**Comment by parv0888** [comment]
> Also please add a unit test.

done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolStatus.java (hunk 1 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolStatus.java:18]: done

File: README.md (hunk 2 above)
Comment (parv0888) [reply README.md:572]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolStatus.java (hunk 1 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolStatus.java:18]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolOpenStatus.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/BakerPoolOpenStatus.java:1]: used, thanks

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PassiveDelegationStatus.java (hunk 6 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PassiveDelegationStatus.java:14]: done

Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PassiveDelegationStatus.java:14]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolInfo.java (hunk 11 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolInfo.java:9]: done, thanks

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java (hunk 12 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java]: removed, added to derived types

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java (hunk 13 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolStatus.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolType.java (hunk 14 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/poolstatus/PoolType.java:6]: done

**Review by EmilLa1 — APPROVED** [review]

**Pull request merged**